### PR TITLE
Align the TLS 1.3 code with coding rules

### PR DESCRIPTION
**Pull request by xffbai**
## Description
Align the TLS 1.3 code with coding rules defined in defined in tls13-experimental.md.

## Status
**READY**

## Backport
Partial backport to 2.x: #5262

### CODE REVIEW

File: include/mbedtls/ssl.h [hunk 1]
```diff
@@ -1173,12 +1173,12 @@ typedef enum
 {
     MBEDTLS_SSL_KEY_EXPORT_TLS12_MASTER_SECRET = 0,
 #if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
-    MBEDTLS_SSL_KEY_EXPORT_TLS13_CLIENT_EARLY_SECRET,
-    MBEDTLS_SSL_KEY_EXPORT_TLS13_EARLY_EXPORTER_SECRET,
-    MBEDTLS_SSL_KEY_EXPORT_TLS13_CLIENT_HANDSHAKE_TRAFFIC_SECRET,
-    MBEDTLS_SSL_KEY_EXPORT_TLS13_SERVER_HANDSHAKE_TRAFFIC_SECRET,
-    MBEDTLS_SSL_KEY_EXPORT_TLS13_CLIENT_APPLICATION_TRAFFIC_SECRET,
-    MBEDTLS_SSL_KEY_EXPORT_TLS13_SERVER_APPLICATION_TRAFFIC_SECRET,
+    MBEDTLS_SSL_KEY_EXPORT_TLS1_3_CLIENT_EARLY_SECRET,
```
Comment (yuhaoth) [inline include/mbedtls/ssl.h:1176]: The name scheme looks different with TLS12. Should we change `MBEDTLS_SSL_KEY_EXPORT_TLS12_MASTER_SECRET` also? @ronald-cron-arm

Comment (ronald-cron-arm) [reply include/mbedtls/ssl.h:1176]: No just stick with TLS 1.3 I think.

File: library/ssl_tls13_keys.h [hunk 2]
```diff
@@ -190,23 +190,23 @@ int mbedtls_ssl_tls1_3_make_traffic_keys(
  * \param dstbuf     The target buffer to write the output of
  *                   \c Derive-Secret to. This must be a writable buffer of
  *                   size \p buflen Bytes.
```
Comment (ronald-cron-arm) [inline library/ssl_tls13_keys.h:192]:
```suggestion
 *                   size \p dstbuf_len Bytes.
```

File: library/ssl_tls13_client.c [hunk 3]
```diff
@@ -205,8 +205,8 @@ static int ssl_tls13_write_supported_groups_ext( mbedtls_ssl_context *ssl,
                                                  size_t *olen )
```
Comment (ronald-cron-arm) [inline library/ssl_tls13_client.c:205]:
```suggestion
                                                 size_t *out_len )
```
To expand your work on length variables I'd suggest to change those as well.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]
This looks almost good to me. Apart from my inline comments there are a couple of remaining "tls1_3" instances that I would prefer to change to "tls13" as well:
```
tests/suites/test_suite_ssl.function:     tls1_3_label_ ## name,                
programs/ssl/ssl_client2.c:#define TLS1_3_VERSION_OPTIONS  ", tls1_3"            
programs/ssl/ssl_server2.c:#define TLS1_3_VERSION_OPTIONS  ", tls1_3"            
library/ssl_tls13_server.c:    MBEDTLS_SSL_DEBUG_MSG( 2, ( "tls1_3 server state: %d", ssl->state ) );
library/ssl_tls13_client.c:    MBEDTLS_SSL_DEBUG_MSG( 2, ( "tls1_3 client state: %d", ssl->state ) );
library/ssl_tls13_keys.c:static const char tls1_3_label_prefix[6] = "tls13 ";   
library/ssl_misc.h:    int tls1_3_kex_modes; /*!< key exchange modes for TLS 1.3 */
library/ssl_misc.h:    } tls1_3_master_secrets;   
```
Regarding TLS1_3_VERSION_OPTIONS, I'd suggest to align (D)TLS 1.2 options as well:
tls1_2 -> tls12 and dtls1_2 ->dtls12.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]
All my comments have been addressed, thanks. From the CI it seems that the changes (d)tls1_2->(d)tls12 need to be done in compat.sh as well though.

**Review by ronald-cron-arm — APPROVED** [review]
LGTM. Please backport to development_2.x the change of option name for ssl_client/server2.c: (d)tls1_2-> (d)tls12.

**Review by yuhaoth — APPROVED** [review]
LGTM

**Pull request merged**